bundle/config/mutator/process_target_mode_test.go

```diff
@@ -69,6 +73,7 @@ func TestProcessTargetModeDevelopment(t *testing.T) {
 	assert.Equal(t, "/Users/lennart.kats@databricks.com/[dev lennart] experiment1", bundle.Config.Resources.Experiments["experiment1"].Name)
 	assert.Equal(t, "[dev lennart] experiment2", bundle.Config.Resources.Experiments["experiment2"].Name)
 	assert.Equal(t, "[dev lennart] model1", bundle.Config.Resources.Models["model1"].Name)
+	assert.Equal(t, "[dev lennart] servingendpoint1", bundle.Config.Resources.ModelServingEndpoints["servingendpoint1"].Name)
```

Comment:
@lennartkats-db FYI, we cannot apply the regular `mode: development` rule because a model serving endpoint's name may only contain alphanumeric, underscores, dashes. We'll need a separate "normalized" prefix to achieve the same. Can you take a look at this?

Reply:
The shortened username shown here already complies to those requirements: (https://github.com/databricks/cli/blob/50b2c0b83bde8bd645c1165ec6b70acf06284151/bundle/config/mutator/populate_current_user.go#L41-L52).

One callout I can make there is that we support all unicode "letters," is that set also supported for endpoints?

Assuming the shortname works, the only thing we'd need here would be a slightly different prefix or suffix at https://github.com/databricks/cli/pull/682/files#diff-9eedd6c63465cd766c89497eda7f212c5b6868054393cfe464d8d1bc69692576R81 in the present PR. For that line, I'd propose to use `"dev_" + b.Config.Workspace.CurrentUser.ShortName + "_"` as the prefix. If we apply that as a prefix (and not as a suffix) then we should be able to support the same style once we support a custom `transform: prefix` option as alluded to in the "deployment modes" design doc.

Reply:
Kk updated the prefix